### PR TITLE
Cheatsheet tests added

### PR DESCRIPTION
**Pull request by Sloff**
Test file for cheat sheets added.

cc @zachthompson @mintsoft @moollaza

### CODE REVIEW

File: lib/DDG/Goodie/CheatSheets.pm [hunk 1]
```diff
@@ -4,6 +4,7 @@ package DDG::Goodie::CheatSheets;
 use JSON::XS;
 use DDG::Goodie;
 use DDP;
+use File::Find::Rule;
```
Comment (mintsoft) [inline lib/DDG/Goodie/CheatSheets.pm]: Is this needed here?

File: t/CheatSheets.t [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            item => 0,
+            options => {
+                content => "DDH.cheat_sheets.detail",
+                moreAt => 0
+            }
+        }
+    };
+}
+
+ddg_goodie_test(
```
Comment (mintsoft) [inline t/CheatSheets.t]: Could/should we iterate over all the cheatsheets in `share` and make sure they all work instead? That way when a new user creates a PR it'd be automatically tested, hopefully it'd catch parse errors in the JSON etc

File: lib/DDG/Goodie/CheatSheets.pm (hunk 1 above)
Comment (zachthompson) [inline lib/DDG/Goodie/CheatSheets.pm]: It is.

File: t/CheatSheets.t (hunk 2 above)
Comment (zachthompson) [inline t/CheatSheets.t]: Great idea.

**Comment by Sloff** [comment]
Hi all, improvements made as suggested

File: t/CheatSheets.t [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        
+        my $defaultName = File::Basename::fileparse($file);
+        $defaultName =~ s/-/ /g;
+        $defaultName =~ s/.json//;
+        
+        $tests{$defaultName." cheat sheet"} = test_zci(build_answer($file));
+        
+        if ($data->{'aliases'}) {
+            foreach my $alias (@{$data->{'aliases'}}) {
+                $tests{$alias." cheat sheet"} = test_zci(build_answer($file));
```
Comment (zachthompson) [inline t/CheatSheets.t]: Pass `$data` to build_answer so you don't need to open the file and decode again.

File: t/CheatSheets.t [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            }
+        }
+    }
+    return %tests;
```
Comment (zachthompson) [inline t/CheatSheets.t]: Return a ref.

**Comment by zachthompson** [comment]
@Sloff Ran it and it works very well! Just needs a couple of tweaks.

**Comment by Sloff** [comment]
Hi @zachthompson, changes made as suggested

**Pull request merged**